content/docs/command-reference/exp/apply.md

```diff
@@ -6,40 +6,89 @@ Put the results from an [experiment](/doc/command-reference/exp) in the
 ## Synopsis
 
 ```usage
-usage: dvc exp apply [-h] [-q | -v] [--no-force] experiment
+usage: dvc exp apply [-h] [-q | -v] experiment
 
 positional arguments:
   experiment     Experiment to be applied
 ```
 
 ## Description
 
-Restores an `experiment` into the workspace, as long as we're on the same
-project baseline (Git `HEAD`) as when the target experiment was run. The
-experiment can be referenced by name or hash (see `dvc exp run` for details).
+Restores an `experiment` into the workspace. The experiment can be referenced by
+name or hash (see `dvc exp run` for details).
 
-Specifically, `dvc exp apply` restores any files or directories needed to
-reflect the experiment conditions and results. This means checking out files
+Specifically, `dvc exp apply` restores any files or directories which exist in
+the experiment, to their exact states from the experiment. This includes files
 tracked both with DVC and Git: code, raw data, <abbr>parameters</abbr>,
 <abbr>metrics</abbr>, resulting artifacts, etc.
 
-⚠️ Conflicting changes in the workspace are overwritten unless `--no-force` is
-used.
-
 This is typically used after choosing a target `experiment` with `dvc exp show`
 or `dvc exp diff`, and before committing it to Git (making it [persistent].
 
 > Note that any [checkpoints] found in the `experiment` will **not** be
 > preserved when applying and committing it. Use `dvc exp branch` instead.
 
+<details>
+
+### Expand for `dvc exp apply` implementation details
+
+`dvc exp apply` is approximately equivalent to the following Git + DVC workflow:
+
+```cli
+$ git stash --include-untracked
+$ git checkout --detach experiment
+$ git reset HEAD^1
+$ git checkout stash@{0} --ours
+$ git stash drop
+$ dvc checkout
+```
+
+The resulting Git + DVC workspace after `dvc exp apply` will contain the
+contents of `experiment`, plus any files from the workspace prior to
+`dvc exp apply` that did not exist in `experiment`.
+
+<admon type="info">
+
+Note that `dvc exp apply` does not actually use the standard Git stash (and does
+not modify `.git/refs/stash`).
+
+</admon>
+
+</details>
+
+<admon type="warn">
+
+Conflicting changes in the workspace are overwritten, but the result of
+`dvc exp apply` can be reverted using Git.
+
+</admon>
+
+<details>
+
+### Expand for details on reverting `dvc exp apply`
+
+`dvc exp apply` can be reverted with the following Git workflow:
+
+```cli
+$ git reset --hard
+$ git stash apply refs/exps/apply/stash
+```
+
+<admon type="info">
+
+Note that this workflow is only valid as long no Git commands which affect
+`HEAD` (such as `git commit` or `git checkout`) have been used since
+`dvc exp apply` was run.
```

Comment:
Couldn't you return to that head to recover even if you've done other things?

Reply:
Yes. You can do

```
$ git reset --hard <original_HEAD>
$ git stash apply refs/exps/apply/stash
```

to recover regardless of what you have done since `dvc exp apply`, but I wasn't sure how much detail we wanted to get into here

Reply:
Agree about the detail. WDYT about making this slightly less alarming since it sounds like you have no way to recover once you make any changes?

```suggestion
Note that `git stash apply` may fail if you run Git commands which affect
`HEAD` (such as `git commit` or `git checkout`) after
`dvc exp apply`.
```